src/Microsoft.Data/PrimitiveColumnContainer.cs

```diff
@@ -81,8 +81,43 @@ public PrimitiveColumnContainer(IEnumerable<T> values)
 
         public PrimitiveColumnContainer(ReadOnlyMemory<byte> buffer, ReadOnlyMemory<byte> nullBitMap, int length, int nullCount)
         {
-            Buffers.Add(new ReadOnlyDataFrameBuffer<T>(buffer, length));
-            NullBitMapBuffers.Add(new ReadOnlyDataFrameBuffer<byte>(nullBitMap, length));
+            ReadOnlyDataFrameBuffer<T> dataBuffer;
+            if (buffer.IsEmpty)
+            {
+                DataFrameBuffer<T> mutableBuffer = new DataFrameBuffer<T>();
+                mutableBuffer.EnsureCapacity(length);
+                mutableBuffer.RawSpan.Fill(default(T));
+                dataBuffer = mutableBuffer;
+            }
+            else
+                dataBuffer = new ReadOnlyDataFrameBuffer<T>(buffer, length);
+            Buffers.Add(dataBuffer);
+            int bitMapBufferLength = (int)Math.Ceiling(length / 8.0);
```

Comment:
```suggestion
            int bitMapBufferLength = (length + 7) / 8;
```
No need for floating points here. 

With `(uint)`s the JIT will emit a bitshift instead of the division -- cf. L217 where we applyed that trick.
But here this will be too much micro-optimization, so plain integer arithmetic will do a good job.